roles/vendor/redis/tasks/install_redis.yml

```diff
@@ -0,0 +1,22 @@
+- name: Download Redis stable
+  get_url:
+    url: https://download.redis.io/releases/redis-{{redis_stable_version}}.tar.gz
+    dest: /opt
+    mode: '0440'
+    checksum: sha256:{{redis_sha256}}
+
+- name: Unarchive redis tarball build
+  unarchive:
+    src: '{{redis_install_dir}}-{{redis_stable_version}}.tar.gz'
+    dest: /opt
+    remote_src: yes
+
+- name: Remove redis tarball
+  ansible.builtin.file:
+    path: '{{redis_install_dir}}-{{redis_stable_version}}.tar.gz'
+    state: absent
+
+- name: Compile Redis
+  make:
+    chdir: '{{redis_install_dir}}-{{redis_stable_version}}'
+    target: install
```

Comment:
I would suggest using the officially supported build from the Redis APT repo. Particularly because it'll allow for unattended updates in case of security issues. https://redis.io/docs/getting-started/installation/install-redis-on-linux/